src/pop/processor.py

```diff
@@ -0,0 +1,149 @@
+        prompt: str
+            The prompt to send to the LLM.
+        uuid: UUID
+            The UUID of the prompt.
+
+        Returns:
+        -------
+        explanation: str
+            The explanation generated by the LLM.
+        """
+
+
+        logger.info(f"Sending prompt with UUID: {uuid} ...")
+
+        try:
+            response = completion(
+                model=f"{self.settings.provider}/{self.settings.model}",
+                base_url=self.settings.url,
+                temperature=self.settings.temperature,
+                messages=[
+                    {"role": "system", "content": self.settings.role},
+                    {"role": "user", "content": prompt},
+                ],
+            )
+
+        except Exception as e:
+            msg = f"""Could not provide a completion:
+            - model: {self.settings.provider}/{self.settings.model}
+            - temperature: {self.settings.temperature}
+            - base_url: {self.settings.url}
+            - role: {self.settings.role},
+            - prompt: {prompt},
+            - error: {e}
+            """
+            logger.error(e)
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail=msg
+            )
+
+        else:
+            logger.info(f"Received response from model {self.settings.model}.")
+            explanation = response.choices[0].message.content
+            return explanation
+
+    def get_prompt(
+        self, type: str, name: str, output: str, description: str
+    ) -> tuple[str, UUID]:
+        """
+        Craft a prompt from the received parameters and store it.
+
+        Parameters:
+        ----------
+        type: str
+            The type of the prompt. Either "host" or "service".
+        output: str
+            The output of the plugin.
+        name: str, optional
+            The name of the host or service.
+        description: str, optional
+            The description of the host or service.
+
+        Returns:
+        -------
+        prompt: str
+            The prompt to send to the LLM.
+        uuid: UUID
+            The UUID of the prompt.
+        """
+
+        uuid = uuid4()
+        prompt = TEMPLATE_PROMPT.format(
+            output=output,
+            type=type,
+            name=name,
+            description=description,
+            length=self.settings.length,
+            language=self.settings.language.value,
+        )
+        self.prompts[uuid] = prompt
+        logger.info(f"Prompt created with UUID: {uuid}.")
+        return prompt, uuid
+
+    def configure(self):
+        """Load params from config file and create one if it doesn't exists."""
+
+        # Default path to the root of the project if not provided.
+        default_path = os.path.join(os.getcwd(), "pop.yaml")
+        path = os.environ.get("POP_CONFIG_PATH", default_path)
+
+        if not os.path.exists(path):
+            with open(path, "w"):
+                pass
```

Comment:
If you want to create an empty file, I think a more straight approach would be to use [`pathlib`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.touch):

```
from pathlib import Path

Path("/path/to/file.txt").touch()
```